corehq/form_processor/management/commands/fix_edit_forms_attachments.py

```diff
@@ -39,17 +51,18 @@ def _find_sql_forms_with_missing_attachments():
                              .filter(operation='edit')
                              .values_list('form_id', flat=True))
             for edit_form_id in with_progress_bar(edit_form_ids):
-                edited_form = form_accessor.get_form(edit_form_id)
-                deprecated_form_id = edited_form.deprecated_form_id
-                original_attachments = list(FormAccessorSQL.get_attachments_for_forms([deprecated_form_id]))
-                new_attachments = list(FormAccessorSQL.get_attachments_for_forms([edit_form_id]))
-                new_attachment_names = [a.name for a in new_attachments]
-                for attachment in original_attachments:
-                    if attachment.name not in new_attachment_names:
-                        assert attachment.form_id == deprecated_form_id
-                        add_forms_with_attachments[edit_form_id].append(
-                            (attachment.attachment_id, deprecated_form_id)
-                        )
+                edit_chain = get_sql_previous_versions(edit_form_id, form_accessor)
+                original_form = edit_chain[0]
+                original_attachments = list(FormAccessorSQL.get_attachments_for_forms([original_form.form_id]))
+                for edited_form in edit_chain[1:]:
+                    attachments = list(FormAccessorSQL.get_attachments_for_forms([edited_form.form_id]))
+                    attachment_names = [a.name for a in attachments]
+                    for attachment in original_attachments:
+                        if attachment.name not in attachment_names:
+                            assert attachment.form_id == original_form.form_id
+                            add_forms_with_attachments[edited_form.form_id].append(
+                                (attachment.attachment_id, original_form.form_id)
```

Comment:
I think you want `attachment.name` here. The semantics of `attachment.attachment_id` are somewhat unclear; I think each attachment (even when copied to new forms) gets a new/unique `attachment_id`. Using `attachment.name` should also make it easier to get the attachment on the form later.

Also, I think it would make sense to pass a reference to `original_form` rather than `original_form.form_id` so you don't have to reload the form and its attachments (which are memoized) when copying attachments. Probably even would make sense to keep a reference to the edited form(s) as well so you don't have to load it again when copying each attachment.

Reply:
updated to pass object instead to avoid any reload of form or the attachment